test/verify/check-machines

```diff
@@ -370,6 +380,71 @@ class TestMachines(MachineCase):
             print("Libvirt version does not support disk statistics")
             b.wait_present("#vm-{0}-disksstats-unsupported".format(name))
 
+    def testLibvirt(self):
+        b = self.browser
+        m = self.machine
+
+        libvirtServiceName = self.getLibvirtName()
+
+        def checkLibvirtEnabled():
+            try:
+                m.execute("systemctl -q is-enabled {0}".format(libvirtServiceName))
+                return True
+            except:  # exception if return code != 0
+                return False
+
+        self.startVm("subVmTest1")
+        self.login_and_go("/machines")
+
+        b.wait_in_text("body", "Virtual Machines")
+        b.wait_present("tbody tr th")
+        b.wait_in_text("tbody tr th", "subVmTest1")
+
+        b.wait(lambda: checkLibvirtEnabled())
+        m.execute("systemctl disable {0}".format(libvirtServiceName))
+        print(self.getLibvirtName())
+        m.execute("systemctl stop {0}".format(libvirtServiceName))
+        b.wait(lambda: not checkLibvirtEnabled())
+        b.wait_present("#slate-header")
+        b.wait_in_text("#slate-header", "Virtualization Service (libvirt) is Not Active")
+        b.wait_present("#enable-libvirt:not(:checked)")
+        b.set_checked("#enable-libvirt", True)
+        b.click("#start-libvirt")
+
+        b.wait_in_text("body", "Virtual Machines")
+        b.wait(lambda: checkLibvirtEnabled())
+        b.wait_present("tbody tr th")
+        b.wait_in_text("tbody tr th", "subVmTest1")
+
+        m.execute("systemctl stop {0}".format(libvirtServiceName))
+        b.wait_present("#slate-header")
+        b.wait_in_text("#slate-header", "Virtualization Service (libvirt) is Not Active")
+        b.wait_present("#enable-libvirt:checked")
+        b.set_checked("#enable-libvirt", False)
+        b.click("#start-libvirt")
+        b.wait(lambda: not checkLibvirtEnabled())
+        b.wait_in_text("body", "Virtual Machines")
+        b.wait_present("tbody tr th")
+        b.wait_in_text("tbody tr th", "subVmTest1")
+
+        m.execute("systemctl enable {0}".format(libvirtServiceName))
+        m.execute("systemctl stop {0}".format(libvirtServiceName))
+        b.wait(lambda: checkLibvirtEnabled())
+        b.wait_present("#slate-header")
+        b.wait_in_text("#slate-header", "Virtualization Service (libvirt) is Not Active")
+        b.wait_present("#enable-libvirt:checked")
+
+        b.click("#troubleshoot")
+        b.leave_page()
+        url_location = "/system/services#/{0}".format(libvirtServiceName)
+        b.wait(lambda: url_location in b.eval_js("window.location.href"))
+
+        # HACK: restarting libvirtd causes this completely unrelated SELinux issue
+        self.allow_journal_messages('.*denied.*search.*"systemd-machine".*dev="proc".*')
+        # enabling/disabling libvirt causes unwanted logs
+        self.allow_journal_messages('.*libvirt.*')
+        self.allow_journal_messages('.*virt.*socket.*')
```

Comment:
Sorry, this looks too broad and can easily hide errors. What are the complete messages here, and how do they break the test? We only check for unexpected log messages on cockpit.service and type=14?? audit messages.

Reply:
I am pretty sure it was complaining about creating/removing symlinks, but now it is fine on my setup. 

It only throws this on ubuntu-1604
``` 
audit: type=1400 audit(1521636424.736:15): apparmor="DENIED" operation="open" profile="libvirt-a73c87e3-37a7-43c0-8671-b477f5783d95" name="/etc/gss/mech.d/" pid=2069 comm="qemu-system-x86" requested_mask="r" denied_mask="r" fsuid=64055 ouid=0
```

Reply:
debian-testing
```
audit: type=1400 audit(1521637589.708:41): apparmor="DENIED" operation="signal" profile="/usr/sbin/libvirtd" pid=1146 comm="libvirtd" requested_mask="send" denied_mask="send" signal=hup peer="unconfined"
```

Reply:
@suomiy : The gss failure is issue #6901, we already have a naughty override for this. We don't yet have one  for the "signal" thing.

Reply:
Removed #6901. Anyway  I think `--sit` should not fail without `--thorough` option on known issue. This got me confused.